File: uva/10070.py
#no heads up about big numbers? cool

import sys

first = True
for line in sys.stdin:
    if not first:
        print("")
    first = False
    y = int(line)
    leap = False
    ordi = True
    if (y%4 == 0 and y%100 != 0) or y%400 == 0:
        leap = True
        ordi = False
        print("This is leap year.")
    if y%15 == 0:
        ordi = False
        print("This is huluculu festival year.")
    if leap and y%55 == 0:
        ordi = False
        print("This is bulukulu festival year.")
    if ordi:
        print("This is an ordinary year.")

File: uva/10070.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    long long N; bool tick = false;
    while (cin >> N) {
        if (tick)
            cout << '\n';
        tick = true;
        bool leap = false, ord = true;
        if (N%4 == 0 && (N%400 == 0 || N%100 != 0)) {
            leap = true;
            ord = false;
            cout << "This is leap year.\n";
        }
        if (N%15 == 0) {
            ord = false;
            cout << "This is huluculu festival year.\n";
        }
        if (leap && N%55 == 0) {
            ord = false;
            cout << "This is bulukulu festival year.\n";
        }
        if (ord) {
            cout << "This is an ordinary year.\n";
        }
    }
    return 0;
}

